src/Microsoft.OData.Client/DataServiceClientException.cs

```diff
@@ -66,53 +58,45 @@ public DataServiceClientException(string message, int statusCode)
         public DataServiceClientException(string message, Exception innerException, int statusCode)
             : base(message, innerException)
         {
-            this.state.StatusCode = statusCode;
+            this.statusCode = statusCode;
+        }
 
-#if !PORTABLELIB
-            this.SerializeObjectState += (sender, e) => e.AddSerializedState(this.state);
-#endif
+        [SecurityPermissionAttribute(SecurityAction.Demand, SerializationFormatter = true)]
+#pragma warning disable CS0628 // New protected member declared in sealed class
+        protected DataServiceClientException(SerializationInfo info, StreamingContext context)

```

Comment:
Can you add a comment explaining why we are adding a protected constructor to a sealed class? Does the serializer call this somehow?

Reply:
from https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.iserializable?view=netcore-3.1:

The ISerializable interface implies a constructor with the signature constructor (SerializationInfo information, StreamingContext context). At deserialization time, the current constructor is called only after the data in the SerializationInfo has been deserialized by the formatter. In general, this constructor should be protected **if the class is not sealed**.

Unclear whether, for a sealed class, it should still be protected. I assume this doesn't work if it's marked as private.

I'm okay with the current code, just please add a comment to the new constructors that they are for serialization and marked protected as per .NET guidelines.

Reply:
@mikepizzo making this constructor private works fine.